### PR TITLE
Upgrade clap to 4.3.9

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: src/cli_util.rs [hunk 1]
```diff
@@ -2279,8 +2279,14 @@ fn handle_early_args(
     args: &[String],
     layered_configs: &mut LayeredConfigs,
 ) -> Result<(), CommandError> {
-    // ignore_errors() bypasses errors like "--help" or missing subcommand
-    let early_matches = app.clone().ignore_errors(true).get_matches_from(args);
```
Comment (chooglen) [inline src/cli_util.rs:2283]: `get_matches_from()` -> `try_get_matches_from()?` makes sense - we should try to propagate the error.

File: src/cli_util.rs [hunk 2]
```diff
@@ -2279,8 +2279,14 @@ fn handle_early_args(
...
+    // ignore_errors() bypasses errors like missing subcommand
+    let early_matches = app
+        .clone()
+        .disable_version_flag(true)
+        .disable_help_flag(true)
```
Comment (chooglen) [inline src/cli_util.rs:2286]: I see this is tested by `tests/test_global_opts.rs:test_early_args()`, which is probably how you found about it too :)

**Review by chooglen — APPROVED** [review]
LGTM. I wonder why clap decided to change the way `ignore_errors()` works.

File: src/cli_util.rs (hunk 2 above)
Comment (martinvonz) [reply src/cli_util.rs:2286]: Exactly, that's what failed in https://github.com/martinvonz/jj/pull/1765

**Comment by martinvonz** [comment]
> LGTM. I wonder why clap decided to change the way `ignore_errors()` works.

It was https://github.com/clap-rs/clap/pull/4988. I didn't read enough to understand it.

**Comment by chooglen** [comment]
I've read all the comments and still don't understand it :)

**Pull request merged**